src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/repository/OffenderCourseAttendanceRepository.kt

```diff
@@ -18,15 +18,16 @@ interface OffenderCourseAttendanceRepository : JpaRepository<OffenderCourseAtten
 
   @Query(
     value = """
-      select new uk.gov.justice.digital.hmpps.nomisprisonerapi.activities.api.BookingCount(oca.offenderBooking.bookingId, count(oca))
+      select new uk.gov.justice.digital.hmpps.nomisprisonerapi.activities.api.BookingCountWithPay(oca.offenderBooking.bookingId, count(oca), sum(otd.payAmount))
       from OffenderCourseAttendance oca
       join OffenderBooking ob on oca.offenderBooking = ob
+      join OffenderTransactionDetail otd on oca.transaction = otd.transaction and oca.eventId = otd.courseAttendance.eventId
       where oca.prison.id = :prisonId
       and oca.eventDate = :date
       and oca.pay = true
       group by oca.offenderBooking.bookingId
       order by oca.offenderBooking.bookingId
     """,
   )
-  fun findBookingPaidAttendanceCountsByPrisonAndDate(prisonId: String, date: LocalDate): List<BookingCount>
+  fun findBookingPaidAttendanceCountsByPrisonAndDate(prisonId: String, date: LocalDate): List<BookingCountWithPay>
```

Comment:
This is the main change - we now return the amount paid as well as the number of attendances recorded so that we can compare the amount pai with DPS during attendance reconciliation.